### PR TITLE
Runtime software versioning data

### PR DESCRIPTION
**Pull request by cnivolle**
## Describe your PR

This PR adds a simple way to **manage available versions of software or runtime** deployed on Clever Cloud.

Yaml files are used to render the list of a runtime or add-on.

So far, on those are made (and the version numbers needs a review): 
- DB: MySQL, MongoDB, PostgreSQL
- Runtime: PHP, Python

### Add-ons

- [MySQL](http://documentation-pr-324.cleverapps.io/doc/addons/mysql/#supported-versions)
- [PG](http://documentation-pr-324.cleverapps.io/doc/addons/postgresql/#supported-versions)
- [MongoDB](http://documentation-pr-324.cleverapps.io/doc/addons/mongodb/#supported-versions)
- [Redis](http://documentation-pr-324.cleverapps.io/doc/addons/redis/#version)
- [Elastic](http://documentation-pr-324.cleverapps.io/doc/addons/elastic/#versions)

### Applications

- [.NET](http://documentation-pr-324.cleverapps.io/doc/applications/dotnet/#net-version)
- [Elixir](http://documentation-pr-324.cleverapps.io/doc/applications/elixir/#mandatory-configuration)
- [Java-Gradle](http://documentation-pr-324.cleverapps.io/doc/applications/java/java-gradle/#available-java-versions)
- [Java-Maven](http://documentation-pr-324.cleverapps.io/doc/applications/java/java-maven/#available-java-versions)
- [Java-Jar](http://documentation-pr-324.cleverapps.io/doc/applications/java/java-jar/#available-java-versions)
- [Java-War](http://documentation-pr-324.cleverapps.io/doc/applications/java/java-war/#available-java-versions)
- [Node.js](http://documentation-pr-324.cleverapps.io/doc/applications/javascript/nodejs/#set-nodejs-version)
- [PHP](http://documentation-pr-324.cleverapps.io/doc/applications/php/#choose-your-php-version)
- [Python](http://documentation-pr-324.cleverapps.io/doc/applications/python/#supported-versions)

## Checklist

- [ ] My PR is related to an opened issue : #
- [x] I've read the [contributing guidelines](/CleverCloud/documentation/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: content/doc/applications/php/_index.md [hunk 1]
```diff
@@ -20,7 +20,9 @@ aliases:
 PHP is a widely-used general-purpose scripting language that is especially suited for Web development and can be embedded
 into HTML.
 
-PHP is available on our platform with the branches 5.6, 7.2, 7.3, 7.4, 8.0, 8.1 and 8.2. You can use FTP or Git to deploy your applications.
+### Supported Versions
+
+The supported versions are {{< runtimes_versions PHP >}}.
```
Comment (davlgd) [inline content/doc/applications/php/_index.md:25]:
```suggestion
The supported versions are: {{< runtimes_versions PHP >}}
```

File: content/doc/applications/python/_index.md [hunk 2]
```diff
@@ -22,7 +22,9 @@ comments: false
 
 Python is a programming language that lets you work more quickly and integrate your systems more efficiently.
 
-Python 2.7, 3.7, 3.8, 3.9, 3.10 and 3.11 are available on Clever Cloud. You can use Git to deploy your application.
+### Supported Versions
+
+The supported versions are {{< runtimes_versions python >}}
```
Comment (davlgd) [inline content/doc/applications/python/_index.md:27]:
```suggestion
The supported versions are: {{< runtimes_versions python >}}
```

File: content/doc/addons/mongodb/_index.md [hunk 3]
```diff
@@ -18,15 +18,15 @@ aliases:
 - /doc/deploy/addon/mongodb/managing-mongodb
 type: docs
 ---
+## Overview
 
 MongoDB is an open source NoSQL document-oriented database. We provide these databases with daily backups and monitoring for both shared and dedicated plans.
 
-## Versions
+### Supported Versions
```
Comment (davlgd) [inline content/doc/addons/mongodb/_index.md:25]:
```suggestion
### Supported versions
```

File: content/doc/addons/mysql/_index.md [hunk 4]
```diff
@@ -14,15 +14,12 @@ aliases:
...
 type: docs
 ---
-
+## Overview
 MySQL is an open-source relational database management system (RDBMS).
 
-## Versions
-
-The version currently installed by the add-on is :
+### Supported Versions
```
Comment (davlgd) [inline content/doc/addons/mysql/_index.md:20]:
```suggestion
### Supported versions
```

File: content/doc/addons/postgresql/_index.md [hunk 5]
```diff
@@ -17,14 +17,14 @@ aliases:
 
 type: docs
 ---
+## Overview
 
 PostgreSQL is an object-relational database management system (ORDBMS) with an emphasis on extensibility and on standards-compliance.
 
-## Versions
+### Supported Versions
```
Comment (davlgd) [inline content/doc/addons/postgresql/_index.md:24]:
```suggestion
### Supported versions
```

File: content/doc/applications/php/_index.md [hunk 6]
```diff
@@ -20,7 +20,9 @@ aliases:
 PHP is a widely-used general-purpose scripting language that is especially suited for Web development and can be embedded
 into HTML.
 
-PHP is available on our platform with the branches 5.6, 7.2, 7.3, 7.4, 8.0, 8.1 and 8.2. You can use FTP or Git to deploy your applications.
+### Supported Versions
```
Comment (davlgd) [inline content/doc/applications/php/_index.md:23]:
```suggestion
### Supported versions
```

File: content/doc/applications/python/_index.md [hunk 7]
```diff
@@ -22,7 +22,9 @@ comments: false
 
 Python is a programming language that lets you work more quickly and integrate your systems more efficiently.
 
-Python 2.7, 3.7, 3.8, 3.9, 3.10 and 3.11 are available on Clever Cloud. You can use Git to deploy your application.
+### Supported Versions
```
Comment (davlgd) [inline content/doc/applications/python/_index.md:25]:
```suggestion
### Supported versions
```

File: static/partials/language-specific-deploy/php.md [hunk 8]
```diff
@@ -2,17 +2,9 @@
 
 ### Choose your PHP version
 
-Set the `CC_PHP_VERSION` environment variable to one of the following values:
+Set the `CC_PHP_VERSION` environment variable to one of the following values: {{< runtimes_versions PHP >}}.
```
Comment (davlgd) [inline static/partials/language-specific-deploy/php.md:5]:
```suggestion
Set the `CC_PHP_VERSION` environment variable to one of the following values: {{< runtimes_versions PHP >}}
```

**Review by davlgd — CHANGES_REQUESTED** [review]
It's a good first iteration for this, thanks for the great job on it!. Just some nitpicks

**Review by davlgd — DISMISSED** [review]

**Review by juliamrch — CHANGES_REQUESTED** [review]
Branch needs to be rebased in order to comply with Hextra 0.8 version

**Comment by cnivolle** [comment]
It has been rebased (🥵) and works with the last version of Hextra.

A few things needs to be done : 
- improve the UI a bit (comas, final dots etc.)
- do a pass on several product pages to make sure we don't miss any product versioning.

File: .github/workflows/review-app.yml [hunk 9]
```diff
@@ -20,7 +20,7 @@ jobs:
           ref: ${{ github.event.pull_request.head.ref }}
           token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0
-      - name: Create review app
+      - name: Manage reveiw app
```
Comment (juliamrch) [inline .github/workflows/review-app.yml:23]:
```suggestion
      - name: Manage review app
```

**Comment by juliamrch** [comment]
Branch is updated and I resolved the few conflicts created this morning (oupsy). Shall we merge this or keep revamping the versioning system for all add-ons and runtimes, and then merge? I can take it from there if needed. What do you think?

File: data/software_versions_shared_dedicated.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    - "v12"
+    - "v13"
+    - "v14"
+    - "v15"
+  dev:
+    - "v15"
+mysql:
+  dedicated:
+    - "v5.7.20"
+    - "v8.0"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:22]:
```suggestion
    - "v8.0.36"
```

File: data/software_versions_shared_dedicated.yml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  dev:
+    - "v8.0"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:24]:
```suggestion
    - "v8.0.22"
```

File: data/software_versions_shared_dedicated.yml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+#    - version_1
+#    - version_1
+#    - version_2
+pg:
+  dedicated:
+    - "v11"
+    - "v12"
+    - "v13"
+    - "v14"
+    - "v15"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:16]:
```suggestion
    - "v15.7"
```

File: data/software_versions_shared_dedicated.yml [hunk 13]
```diff
@@ -0,0 +1,29 @@
+# Example usage:
+#myDB
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:2]: It misses Redis and Elasticsearch: 
elasticsearch: 
   dedicated:
       - "v7.17.10"
       - "v8.10.2"
 redis: 
   dedicated:
       - "v7.2.4"

File: data/software_versions_shared_dedicated.yml [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+#  dedicated:
+#    - version_1
+#    - version_1
+#    - version_2
+pg:
+  dedicated:
+    - "v11"
+    - "v12"
+    - "v13"
+    - "v14"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:15]:
```suggestion
    - "v14.12"
```

File: data/software_versions_shared_dedicated.yml [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+#    - version_2
+#  dedicated:
+#    - version_1
+#    - version_1
+#    - version_2
+pg:
+  dedicated:
+    - "v11"
+    - "v12"
+    - "v13"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:14]:
```suggestion
    - "v13.15"
```

File: data/software_versions_shared_dedicated.yml [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+#    - version_1
+#    - version_2
+#  dedicated:
+#    - version_1
+#    - version_1
+#    - version_2
+pg:
+  dedicated:
+    - "v11"
+    - "v12"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:13]:
```suggestion
    - "v12.19"
```

File: data/software_versions_shared_dedicated.yml [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+#  shared:
+#    - version_1
+#    - version_2
+#  dedicated:
+#    - version_1
+#    - version_1
+#    - version_2
+pg:
+  dedicated:
+    - "v11"
```
Comment (aurrelhebert) [inline data/software_versions_shared_dedicated.yml:12]:
```suggestion
    - "v11.22"
```

**Review by aurrelhebert — COMMENTED** [review]
LGTM

**Comment by juliamrch** [comment]
## Follow up work: 

### Add-ons

- [ ] Add add-ons versions data
  - [ ] Elastic
  - [ ] Redis 
- [ ] Replace with shortcodes

### Applications

- [ ] Add runtime versions data
  - [ ] .NET
  - [ ] Elixir
  - [ ] Java
- [ ] Replace with shortcodes in pages
- [ ] Rework partial to erase **Shared** when not needed in page

File: layouts/shortcodes/software_versions_shared_dedicated.html [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+            <!-- If the index is outside the dev values -->
+          {{ end }}
+        </td>
+      {{ end }}
+    </tr>
+    {{ end }}
+  </tbody>
+</table>
+{{ if $has_dev }}
+  {{ $.Page.RenderString "<!-- markdown -->{{< callout type=\"warning\">}}  <strong>Important</strong>: Dev plans are free and intended <u>solely for testing purposes</u>. They do not provide the same guarantees or SLAs as dedicated plans. {{< /callout >}}" | safeHTML }}
```
Comment (juliamrch) [inline layouts/shortcodes/software_versions_shared_dedicated.html:36]:
```suggestion
  {{ $.Page.RenderString "<!-- markdown -->{{< callout type=\"warning\">}}  <strong>Important</strong>: Dev plans are free and intended <u>solely for testing purposes</u>. They don't provide the same guarantees or SLAs as dedicated plans. {{< /callout >}}" | safeHTML }}
```

**Review by juliamrch — DISMISSED** [review]
wonderful

**Pull request merged**